.circleci/config.yml

```diff
@@ -2,65 +2,271 @@ version: 2.1
+    resource_class: xlarge
+    steps:
+      - checkout
+      - setup_remote_docker:
+          docker_layer_caching: true
+      - run: sudo chown -R circleci:circleci /usr/local/bin
+      - run: sudo chown -R circleci:circleci /usr/local/lib/python3.7/site-packages
+      - restore_cache:
+          name: Restore Pip Cache
+          keys:
+            - pip-packages-v5-{{ .Branch }}-{{ checksum "Pipfile.lock" }}
+            - pip-packages-v5-{{ .Branch }}-
+            - pip-packages-v5-
+      - run:
+          name: install
+          command: |
+            pipenv install --dev
+
+      - save_cache:
+          name: Save Pip Cache
+          key: pip-packages-v5-{{ .Branch }}-{{ checksum "Pipfile.lock" }}
+          paths:
+            - ~/.local
+            - ~/.cache
+      - run:
+          name: create data model
+          command: |
+            make resetdb
+      - run:
+          name: test-server # server tests
+          command: |
+            pipenv run pytest --cov=.
+
+  build-and-test-client:
     docker:
       - image: circleci/python:3.7.3-node-browsers
         environment:
           DATABASE_URL: postgresql://root@localhost:5432/arlo-test
           FLASK_ENV: test
-      - image: circleci/postgres:9.6.2-alpine
+      - image: circleci/postgres:9.6.3-alpine-ram
         environment:
           POSTGRES_USER: root
           POSTGRES_DB: arlo-test
+    resource_class: xlarge
     steps:
       - checkout
+      - setup_remote_docker:
+          docker_layer_caching: true
+      - run: sudo chown -R circleci:circleci /usr/local/bin
+      - run: sudo chown -R circleci:circleci /usr/local/lib/python3.7/site-packages
+      - restore_cache:
+          name: Restore Yarn Cache
+          keys:
+            - yarn-packages-v3-{{ .Branch }}-{{ checksum "yarn.lock" }}
+      - restore_cache:
+          name: Restore Node Cache
+          keys:
+            - npm-packages-v8-{{ .Branch }}-{{ checksum "package-lock.json" }}
       - run:
           name: install
           command: |
-            make install install-development
+            yarn install
+            yarn --cwd client install
+            yarn --cwd client build
+
+      - save_cache:
+          name: Save Yarn Cache
+          key: yarn-packages-v3-{{ .Branch }}-{{ checksum "yarn.lock" }}
+          paths:
+            - ~/.cache/yarn
+
+      - save_cache:
+          name: Save Node Cache
+          key: npm-packages-v8-{{ .Branch }}-{{ checksum "package-lock.json" }}
+          paths:
+            - node_modules
+            - client/node_modules
+
       - run:
           name: Install Chrome
           command: |
             curl -L -o google-chrome.deb https://dl.google.com/linux/direct/google-chrome-stable_current_amd64.deb
             sudo dpkg -i google-chrome.deb
             sudo sed -i 's|HERE/chrome\"|HERE/chrome\" --disable-setuid-sandbox|g' /opt/google/chrome/google-chrome
             rm google-chrome.deb
+
```

Comment:
do we actually need chrome for the frontend tests?

Reply:
I can check, but I tried removing it and it broke stuff. The docker we're loading actually has browsers in it, but yarn complained about their version so the manual install seems necessary.